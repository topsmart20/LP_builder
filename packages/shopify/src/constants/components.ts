import { BuilderElement } from "@builder.io/sdk";
import { Options } from "../interfaces/options";

export interface ComponentInfo {
  component: (block: BuilderElement, options: Options, attributes?: string) => string;
  noWrap?: boolean;
  name: string;
}

export function component(info: ComponentInfo) {
  components[info.name] = info;
  return info.component;
}

export function getComponentInfo(name: string) {
  return components[name];
}

export function getComponent(name: string) {
  const info = getComponentInfo(name)
  return info && info.component;
}

export const components: { [key: string]: ComponentInfo | undefined } = {};
