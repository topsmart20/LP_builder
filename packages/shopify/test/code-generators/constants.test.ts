import { sizes } from '../../src/constants/sizes';

describe('Constants', () => {
  test('sizes are consistent', () => {
    const { xsmall, small, medium, large } = sizes;

    expect(xsmall).toMatchSnapshot();
    expect(small).toMatchSnapshot();
    expect(medium).toMatchSnapshot();
    expect(large).toMatchSnapshot();
  });

  test('getWidthForSize', () => {
    const testSize = sizes.getSizeForWidth(235);
    expect(testSize).toMatchSnapshot();
  });
});
