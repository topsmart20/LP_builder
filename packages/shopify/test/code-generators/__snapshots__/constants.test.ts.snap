// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Constants getWidthForSize 1`] = `"small"`;

exports[`Constants sizes are consistent 1`] = `
Object {
  "default": 0,
  "max": 0,
  "min": 0,
}
`;

exports[`Constants sizes are consistent 2`] = `
Object {
  "default": 321,
  "max": 640,
  "min": 320,
}
`;

exports[`Constants sizes are consistent 3`] = `
Object {
  "default": 642,
  "max": 991,
  "min": 641,
}
`;

exports[`Constants sizes are consistent 4`] = `
Object {
  "default": 991,
  "max": 1200,
  "min": 990,
}
`;
